packages/headless/src/features/commerce/recommendation/recommendation-actions.ts

```diff
@@ -0,0 +1,59 @@
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {AsyncThunkCommerceOptions} from '../../../api/commerce/commerce-api-client';
+import {CommerceRecommendationRequest} from '../../../api/commerce/recommendation/recommendation-request';
+import {isErrorResponse} from '../../../api/search/search-api-client';
+import {nonEmptyString, validatePayload} from '../../../utils/validate-payload';
+import {logQueryError} from '../../search/search-analytics-actions';
+import {
+  buildCommerceAPIRequest,
+  QueryRecommendationCommerceAPIThunkReturn,
+  StateNeededByQueryCommerceAPI,
+} from '../common/actions';
+
+const buildRecommendationCommerceAPIRequest = async (
+  slotId: string,
+  state: StateNeededByQueryCommerceAPI
+): Promise<CommerceRecommendationRequest> => {
+  const commerceAPIRequest = await buildCommerceAPIRequest(state);
+  delete commerceAPIRequest.facets;
```

Comment:
@Spuffynism could we refactor that `buildCommerceAPIRequest` so facets are optional, like that case of sort?

Reply:
Yeah, good idea! I'd prefer that to `delete`ing the facets here.